src/response.jl

```diff
@@ -2,78 +2,71 @@
 # License is MIT: see https://github.com/JuliaFEM/JuliaFEM.jl/blob/master/LICENSE.md
 using Tensors
 
-generate_elastic_tensor{P<:Elastic}(mat::P, dim) = generate_elastic_tensor(mat, dim, nothing)
-
-"""
-    generate_elastic_tensor_2D(E, nu, ::Type{Val{:plane_stress}})
-
-Calculate elastic moduli in plain stress formulation
-
-"""
-function generate_elastic_tensor_2D(E, nu, ::Type{Val{:plane_stress}})
-    D = E/(1.0 - nu^2) .* [
-            1.0   nu    0.0
-             nu  1.0    0.0
-            0.0  0.0 1.0-nu]
-    Tensor{2, 3}(D)
-end
-
-"""
-    generate_elastic_tensor_2D(E, nu, ::Type{Val{:plane_strain}})
-
-Calculate elastic moduli in plane stress formulation
-"""
-function generate_elastic_tensor_2D(E, nu, ::Type{Val{:plane_strain}})
-    D = E/((1.0+nu)*(1.0-2.0*nu)) .* [
-            1.0-nu      nu         0.0
-                nu  1.0-nu         0.0
-                0.0     0.0 1.0-2.0*nu]
-    Tensor{2, 3}(D)
-end
+kron_delta = (i, j) -> i == j ? 1 : 0
+ident_func = (i,j,k,l) -> 1/2 * (kron_delta(i,k)*kron_delta(j,l) + kron_delta(i,l)*kron_delta(j,k))
```

Comment:
This should be `const` or defined not using anonymous functions. Just change to `ident_func(i,j,k,l) = ...` and it will be faster.